docs/plugin-write.md

```diff
@@ -0,0 +1,201 @@
+---
+id: plugin-write
+title: Writing Plugins
+---
+
+## Description
+
+Lets take a closer look at the minimal plugin instance we created in the [intro](/docs/plugin) part:
+
+```golang
+// Plugin is plugin instance
+type Plugin struct{}
+
+// Enable implements plugin.Plugin
+func (c *Plugin) Enable() error {
+	return nil
+}
+
+// Disable implements plugin.Plugin
+func (c *Plugin) Disable() error {
+	return nil
+}
+```
+
+Now it has only implemented the base [`Plugin`](https://godoc.org/github.com/gotify/plugin-api#Plugin) interface, and does nothing at all. To give it more functionalities, we need to implement more interfaces on `*Plugin`.
+
+## APIs
+
+### Intro
+
+The APIs are provided as interfaces and has generally two forms:
+
+The first form is called after plugins are started(after `Enabled` is called). They require the plugin to implements some methods which is called on user request to update information or provide feedbacks. An example is [`Displayer`](https://godoc.org/github.com/gotify/plugin-api#Displayer), which obtains an instruction generated by the plugin, which is then relayed to the user by gotify.
+
+The second form is called before plugins are started(before `Enabled` is called). The methods are in the form of `^Register(.+)Handler$`, which plugins will implement to get a callback to the gotify engine to realize some functionalities. An example is [`Messenger`](https://godoc.org/github.com/gotify/plugin-api#Messenger), which is called with a callback that can be called at any time to send a message to the user.
```

Comment:
It would be great, if you could add linebreaks after the end of a sentence. Then the lines will not be that long